src/utils/hooks/__tests__/useStorage.test.ts

```diff
@@ -84,4 +85,72 @@ describe('useStorage', () => {
 
     expect(getCurrentValue()).toBeNull();
   });
+
+  test('Should validate a string union without a json parse', async () => {
+    const fruit = z.enum(['apple', 'orange', 'banana']);
+
+    const { result } = renderHook(() =>
+      useStorage({
+        defaultValue: 'apple',
+        key: 'people-array',
+        schema: fruit,
+        type: 'local',
+      }),
+    );
+
+    const getCurrentValue = () => result.current.value;
+
+    // Assert initial state
+    const { setValue, value: initialValue } = result.current;
+
+    expect(initialValue).toBe('apple');
+
+    await act(async () => {
+      await setValue('banana');
+    });
+
+    // Assert the updated state
+    expect(getCurrentValue()).toBe('banana');
+
+    // Manually mangle local storage
+    await act(async () => {
+      await setValue('ooga-booga' as unknown as z.infer<typeof fruit>);
+    });
+
+    expect(getCurrentValue()).toBe('apple');
+  });
+
+  test('Should parse a boolean without a schema', async () => {
+    const { result } = renderHook(() =>
+      useStorage({
+        defaultValue: false,
+        json: true,
+        key: 'boolean',
+        type: 'session',
+      }),
+    );
+
+    const getCurrentValue = () => result.current.value;
+
+    // Assert initial state
+    const { setValue, value: initialValue } = result.current;
+
+    expect(initialValue).toBe(false);
+
+    await act(async () => {
+      await setValue(true);
+    });
+
+    // Assert the updated state
+    expect(getCurrentValue()).toBe(true);
+
+    // Manually mangle local storage
+    await act(async () => {
+      await setValue('ooga-booga' as unknown as boolean);
+    });
+
+    // We have no schema, so it should just return the raw JSON parsed value
+    // Not sure why anyone would want this...
+    expect(getCurrentValue()).toBe('ooga-booga');
```

Comment:
I am open to suggestion here. I can't imagine a good use case where you would want to parse JSON but not use a schema, as it would open it up to incorrect type casting.

If I were to remove that case, it seems like it would just make it simpler that if a schema is present, it will always use JSON to set and get.

And with that being said, it might be even better to enforce a schema to mitigate any possible bad data that could get put in storage

Reply:
I agree, when json is set, it should require schema. I also think it should always require schema since there is not a case where we don't need type safe, especially for local and session storage keys 🤔. But requiring it now will also add more maintenance debt. In my opinion, Amino has `ts-toolbelt` as a dev dependencies, we can utilize the `Union.Strict` to make `schema` required when `json` is true for now, and we will convert it all as we go 🤔. What do you think?

Reply:
The amino hook isn't actually used that much yet so I think it is fine to require schema now.